packages/block-editor/src/hooks/metadata.js

```diff
@@ -7,6 +7,10 @@ import { getBlockSupport } from '@wordpress/blocks';
 const META_ATTRIBUTE_NAME = 'metadata';
 
 export function hasBlockMetadataSupport( blockType, feature = '' ) {
+	// Only core blocks are allowed to use __experimentalMetadata until the fetaure is stablised.
+	if ( ! blockType.startsWith( 'core/' ) ) {
+		return false;
+	}
```

Comment:
@ntsekouras This is based on what Adam said in https://github.com/WordPress/gutenberg/issues/47196#issuecomment-1415734319.

I wonder instead if we should simply return `false` based on an "is gutenberg plugin" check?

Reply:
Since we're not landing the functionality in `6.2` do we even need any check?  Experimental APIs can be removed/updated without a warning.

It seems this code is already in `6.1` packages, but since there is no handling for them we don't need to worry about back compat. 🤔 --cc @adamziel

Reply:
> since there is no handling for them

I don't follow this part...

Reply:
It may not be necessary but this seems to be a useful safeguard that prevents other blocks from using this.

Reply:
> since there is no handling for them

I don't follow this part either @ntsekouras.

> I wonder instead if we should simply return false based on an "is gutenberg plugin" check?

@getdave if your goal is to exclude this API from WordPress 6.2 entirely, then sure. If it is to release it in 6.2 but only for core blocks, then I'd stick with `blockType.startsWith( 'core/' )`

Reply:
The code hasn't been used yet so it's difficult. I'd prefer it wasn't release _at all_.

That said, Nik said:

> It seems this code is already in 6.1 packages

So knowing that I don't think we can exclude it entirely.

Therefore shall we just go for limiting to Core blocks? Does that sound reasonable?

Reply:
It does to me :+1:

Reply:
I meant if a block adds `"__experimentalMetadata": true,` to its block.json, does it do anything?

Reply:
Either way, the check is harmless.